derrida/books/views.py

```diff
@@ -543,23 +556,61 @@ def get_proxy_url(self, *args, **kwargs):
         if not canvas:
             raise Http404
 
-        if kwargs['mode'] == 'thumbnail':
-            return canvas.image.thumbnail()
+        mode = kwargs['mode']
+
+        if mode == 'info':
+            return canvas.image.info()
+        elif mode == 'iiif':
+            # also restrict iiif tiles based on large image permission
+            if not instance.allow_canvas_large_image(canvas):
+                raise Http404
+            return canvas.image.info().replace('info.json', kwargs['url'].strip('/'))
 
-        if kwargs['mode'] == 'large':
+        # if large image is requested, make sure it is allowed before
+        # any further processing
+        if mode == 'large':
             # only allow large images for insertions, overview images,
             # and pages with documented interventions
             # - also checks if an image has been suppressed
             if not instance.allow_canvas_large_image(canvas):
                 raise Http404
 
-            return canvas.image.size(height=850, width=850,
-                exact=True)    # exact = preserve aspect
+        # for specific sizes, request image info to determine available
+        # preset sizes and use the closest size larger than what we need
+        # (if the server supports it and provides sizes)
+        if mode in ['thumbnail', 'large', 'smthumb']:
+            resp = requests.get(canvas.image.info())
+            available_sizes = resp.json().get('sizes', [])
+
+        min_width = min_height = None
+        if mode == 'smthumb':
```

Comment:
`if`...`if`...`elif` feels weird to me, but maybe that's a python thing? seems like this could be a `switch`/`case`. but the logic should work either way

Reply:
Yeah, no `switch`/`case` in python - but you're right, that second `if` should be an `elif`! Thanks for catching, I will push a fix for that right away.